cmd/api/src/services/fileupload/file_upload.go

```diff
@@ -82,13 +85,22 @@ func GetFileUploadJobByID(db FileUploadData, jobID int64) (model.FileUploadJob,
 	return db.GetFileUploadJob(jobID)
 }
 
-func SaveToTempFile(location string, fileData io.ReadCloser) (string, error) {
-	if tempFile, err := os.CreateTemp(location, "bh"); err != nil {
-		return "", fmt.Errorf("error creating temp file: %w", err)
-	} else if _, err := io.Copy(tempFile, fileData); err != nil {
-		return "", fmt.Errorf("error copying temp file: %w", err)
+func SaveIngestFile(location string, fileData io.ReadCloser) (string, error) {
+	var typeBuf = make([]byte, 512)
+	tempFile, err := os.CreateTemp(location, "bh")
+	if err != nil {
+		return "", fmt.Errorf("error creating ingest file: %w", err)
+	}
+	defer tempFile.Close()
```

Comment:
We are closing the file with the moved defer here, which is good, but we aren't deleting the invalid temporary file if an error occurs.

Reply:
We have an orphan removal process that takes care of this